File: src/main/java/com/iot/temperature/repository/TemperatureRepository.java
package com.iot.temperature.repository;

import com.iot.temperature.model.Temperature;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.CrudRepository;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

public interface TemperatureRepository extends CrudRepository<Temperature, Long> {

    @Query(value = "SELECT * FROM iot.temperature ORDER BY id DESC LIMIT ?1 ", nativeQuery = true)
    List<Temperature> findLastTemperatures(int duration);

    List<Temperature> findLastTemperaturesByTimestamp(LocalDateTime date);

    List<Temperature> findTemperaturesByTimestampBetween(LocalDateTime start, LocalDateTime end);
}

File: src/main/kotlin/temperature/repository/TemperatureRepository.kt
package temperature.repository

import temperature.model.Temperature
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.CrudRepository
import java.time.LocalDateTime

interface TemperatureRepository : CrudRepository<Temperature?, Long?> {
    @Query(value = "SELECT * FROM iot.temperature ORDER BY id DESC LIMIT ?1 ", nativeQuery = true)
    fun findLastTemperatures(duration: Int): List<Temperature?>?

    fun findLastTemperaturesByTimestamp(date: LocalDateTime?): List<Temperature?>?
    fun findTemperaturesByTimestampBetween(start: LocalDateTime?, end: LocalDateTime?): List<Temperature?>?
    fun findTopByOrderByIdDesc(): Temperature? //    User findTopByOrderByAgeDesc();
}